pkg/controller/signup_check.go

```diff
@@ -0,0 +1,97 @@
+package controller
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/codeready-toolchain/registration-service/pkg/configuration"
+	"github.com/codeready-toolchain/registration-service/pkg/errors"
+	"github.com/gin-gonic/gin"
+)
+
+const (
+	// SignupStatePending represents a signup in state pending approval.
+	SignupStatePending = "pendingApproval"
+	// SignupStateProvisioning represents a signup in state in provisioning.
+	SignupStateProvisioning = "provisioning"
+	// SignupStateProvisioned represents a signup in state provisioned/ready.
+	SignupStateProvisioned = "provisioned"
+)
+
+// SignupCheck implements the SignupCheck endpoint.
+type SignupCheck struct {
+	config *configuration.Registry
+	logger *log.Logger
+}
+
+// SignupCheckPayload payload
+type SignupCheckPayload struct {
+	Ready   bool   `json:"ready"`
+	Reason  string `json:"reason"`
+	Message string `json:"message"`
+}
+
+// NewSignupCheck returns a new SignupCheck instance.
+func NewSignupCheck(logger *log.Logger, config *configuration.Registry) *SignupCheck {
+	return &SignupCheck{
+		logger: logger,
+		config: config,
+	}
+}
+
+var testRequestTimestamp int64
+
+// getTestSignupCheckInfo retrieves a test check info. Used only for tests.
+// It reports provisioning/not ready for 5s, then reports state complete.
+func (hc *SignupCheck) getTestSignupCheckInfo() *SignupCheckPayload {
```

Comment:
what about passing this func as an argument in the `NewSignupCheck`? In that case, this will move to the `_test.go` file and not be part of the "production" code at all. And you'll also be able to get rid of the `if hc.config.IsTestingMode()` condition in  https://github.com/codeready-toolchain/registration-service/pull/41/files#diff-05008956dde833291afa7105f4e18c3aR83-R87

Something like:

```
func NewSignupCheck(logger *log.Logger, config *configuration.Registry, checker func() *SignupCheckPayload) *SignupCheck {
...
}
```

Reply:
Cool idea. Result in 522d208.